xchange-gdax/src/main/java/info/bitrich/xchangestream/gdax/GDAXStreamingMarketDataService.java

```diff
@@ -142,7 +152,11 @@ private boolean containsPair(List<CurrencyPair> pairs, CurrencyPair pair) {
                 .filter(message -> !isNullOrEmpty(message.getType()) && message.getType().equals("match") &&
                         message.getProductId().equals(channelName))
                 .map(s -> {
-                            Trades adaptedTrades = adaptTrades(new GDAXTrade[]{s.toGDAXTrade()}, currencyPair);
+                            Trades adaptedTrades = null;
+                            if ( s.getUserId() != null )
+                                adaptedTrades = adaptTradeHistory(new GDAXFill[]{s.toGDAXFill()});
```

Comment:
I'm working on something similar for Binance right now, and this bit gave me pause.  Are you getting duplicated trades appearing where you get both the `Trade` from the `matches` channel and the `UserTrade` from the `user` channel?

I certainly do with Binance and it itched.  The only way I could think of to resolve the issue looks something like this:

```
        	// To avoid duplication of user and public trades, we delay the public trades by one second
        	// and skip any trade ids which have recently been streamed as user trades. This isn't
        	// perfect, of course...
        	Cache<String, ? super Trade> recentTrades = CacheBuilder.newBuilder().maximumSize(50).build();
        	return publicTrades
        		.concatMap(t -> Observable.just(t).delay(1, TimeUnit.SECONDS))
        		.skipWhile(t -> recentTrades.asMap().containsKey(t.getId()))
        		.mergeWith(
        			getUserTrades()
                		.filter(t -> t.getCurrencyPair().equals(currencyPair))
                		.doOnNext(t -> recentTrades.put(t.getId(), t))
        		);
```

Reply:
I think the code replaces the Trade object with a UserTrade object so there wouldn't be a duplicate.

Honestly in my usage (external to this project) I'm just looking for the fills coming in (UserTrade) so it never hurt my bot if there was or wasn't a Trade reported via the regular trade feed.

I suppose if the record came in via the trade feed with no user information there would be a duplicate.